ui/components/multichain/pages/all-connections/all-connections.js

```diff
@@ -0,0 +1,183 @@
+import React from 'react';
+import { useHistory } from 'react-router-dom';
+import { useSelector } from 'react-redux';
+import { Header, Page } from '../page';
+import {
+  ButtonIcon,
+  ButtonIconSize,
+  IconName,
+  Text,
+} from '../../../component-library';
+import { useI18nContext } from '../../../../hooks/useI18nContext';
+import {
+  BackgroundColor,
+  Color,
+  TextAlign,
+  TextVariant,
+} from '../../../../helpers/constants/design-system';
+import { DEFAULT_ROUTE } from '../../../../helpers/constants/routes';
+import {
+  getAllConnectedAccounts,
+  getConnectedSubjectsForAllAddresses,
+  getSnapsList,
+} from '../../../../selectors';
+import { Tab, Tabs } from '../../../ui/tabs';
+import { ConnectionListItem } from './connection-list-item';
+
+export const AllConnections = () => {
+  const t = useI18nContext();
+  const history = useHistory();
+  const TABS_THRESHOLD = 5;
+  let totalConnections = 0;
+  const connectedSubjectsForAllAddresses = useSelector(
+    getConnectedSubjectsForAllAddresses,
+  );
+  const connectedAddresses = useSelector(getAllConnectedAccounts);
+  const connectedSiteData = {};
+  connectedAddresses.forEach((connectedAddress) => {
+    connectedSubjectsForAllAddresses[connectedAddress].forEach((app) => {
+      if (!connectedSiteData[app.origin]) {
+        connectedSiteData[app.origin] = { ...app, addresses: [] };
+      }
+      connectedSiteData[app.origin].addresses.push(connectedAddress);
+    });
+  });
+
+  const sitesConnectionsList = {};
+  Object.keys(connectedSiteData).forEach((siteKey) => {
+    const siteData = connectedSiteData[siteKey];
+    const { name, iconUrl, origin, subjectType, extensionId, addresses } =
+      siteData;
+
+    if (!sitesConnectionsList[name]) {
+      sitesConnectionsList[name] = {
+        name,
+        iconUrl,
+        origin,
+        subjectType,
+        extensionId,
+        addresses: [],
+      };
+      totalConnections += 1;
+    }
+
+    sitesConnectionsList[name].addresses.push(...addresses);
+  });
+
+  const snapsConnectionsList = {};
+  const connectedSnapsData = useSelector(getSnapsList);
+  Object.keys(connectedSnapsData).forEach((snap) => {
+    const snapData = connectedSnapsData[snap];
+    const { id, name, packageName, iconUrl, subjectType } = snapData;
+
+    if (!snapsConnectionsList[name]) {
+      snapsConnectionsList[name] = {
+        id,
+        name,
+        iconUrl,
+        packageName,
+        subjectType,
+      };
+      totalConnections += 1;
+    }
+  });
+
+  const shouldShowTabsView =
+    totalConnections > TABS_THRESHOLD &&
+    Object.keys(sitesConnectionsList).length > 0 &&
+    Object.keys(snapsConnectionsList).length > 0;
+
+  const handleConnectionClick = (connection) => {
+    // TODO: go to connection details page
+    console.log('connection clicked: ', connection);
+  };
+  const connectionsList = (connectionList) =>
```

Comment:
Suggestion: change the name to `renderConnectionsList` to indicate this is a DOM return result